common.mk

```diff
@@ -207,27 +207,21 @@ linux-cleaner-common: linux-defconfig-clean
 ################################################################################
 # EDK2 / Tianocore
 ################################################################################
-# Make sure edksetup.sh only will be called once and that we don't rebuild
-# BaseTools again and again.
-$(EDK2_PATH)/Conf/target.txt:
-	set -e && cd $(EDK2_PATH) && source edksetup.sh && \
-	$(MAKE) -j1 -C $(EDK2_PATH)/BaseTools
-
 .PHONY: edk2-common
-edk2-common: $(EDK2_PATH)/Conf/target.txt
-	set -e && cd $(EDK2_PATH) && source edksetup.sh && \
-	$(call edk2-call)
+edk2-common:
+	export WORKSPACE=$(ROOT) && \
+	export PACKAGES_PATH=$(EDK2_PATH):$(ROOT)/edk2-platforms && \
+	source $(EDK2_PATH)/edksetup.sh && \
+	$(MAKE) -j1 -C $(EDK2_PATH)/BaseTools && \
+	$(call edk2-call) all
 
 .PHONY: edk2-clean-common
 edk2-clean-common:
+	export WORKSPACE=$(ROOT) && \
+	export PACKAGES_PATH=$(EDK2_PATH):$(ROOT)/edk2-platforms && \
 	set -e && cd $(EDK2_PATH) && source edksetup.sh && \
```

Comment:
minor: `set -e && cd $(EDK2_PATH)` needed?  sync with `source $(EDK2_PATH)/edksetup.sh` above?

Reply:
You're meaning the `set -e`? Yeah that is unneccesary with all the `&&` being used. Is that what you had in mind?

Reply:
it is just a minor comment to align `edk2-clean-common` sequence on `edk2-common` when can be similar.

Reply:
Also, don't see `WORKSPACE` and `PACKAGES_PATH` used anywhere else. Are they required for the build to build?

Reply:
I guess so, the `build` script in UEFI seems to use those.